odoo/addons/base/data/res_users_demo.xml

```diff
@@ -42,10 +56,24 @@
             <field name="image_1920" type="base64" file="base/static/img/user_demo-image.png"/>
         </record>
 
+        <record id="user_restricted" model="res.users">
```

Comment:
Why not use `partner_demo` and give them the necessary permissions for the test?